### PR TITLE
BUG: Fix import for new versions of ibis

### PR DESCRIPTION
**Pull request by gerrymanoim**
The compiler is now at https://github.com/ibis-project/ibis/blob/master/ibis/backends/base_sql/compiler.py

### CODE REVIEW

**Comment by jreback** [comment]
@tswast any issues here?

**Comment by jreback** [comment]
cc @gerrymanoim https://github.com/ibis-project/ibis-bigquery/pull/59/checks?check_run_id=2630857440

**Comment by tswast** [comment]
I'd also like to make sure we maintain compatibility with ibis 1.4.0. At least until there are a few 2.x releases. I see there are some failing test in that build. https://github.com/ibis-project/ibis-bigquery/pull/59/checks?check_run_id=2630857546

e.g.

```
Run python -m pytest tests/unit
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/runner/work/ibis-bigquery/ibis-bigquery
collected 2 items / 4 errors

==================================== ERRORS ====================================
__________________ ERROR collecting tests/unit/test_client.py __________________
tests/unit/test_client.py:4: in <module>
    import ibis_bigquery.client
ibis_bigquery/__init__.py:11: in <module>
    from .client import (BigQueryClient, BigQueryDatabase, BigQueryQuery,
ibis_bigquery/client.py:28: in <module>
    from . import compiler as comp
ibis_bigquery/compiler.py:71: in <module>
    class BigQuerySelectBuilder(comp.SelectBuilder):
E   AttributeError: module 'ibis.backends.base_sql.compiler' has no attribute 'SelectBuilder'
_________________ ERROR collecting tests/unit/test_compiler.py _________________
```

File: ibis_bigquery/compiler.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 import ibis
 
 try:
-    import ibis.backends.base_sqlalchemy.compiler as comp
```
Comment (tswast) [inline ibis_bigquery/compiler.py:10]: Based on the test failures, this is the name in 1.4.0.

Let's add another try-except block and fall back to that.

**Comment by gerrymanoim** [comment]
Sorry about that - I added another try/except so that the CI now passes: https://github.com/gerrymanoim/ibis-bigquery/pull/1. 

If I understand the github actions file, `github` should test against head of ibis master?

**Review by jreback — APPROVED** [review]
lgtm

**Comment by jreback** [comment]
cc @tswast

**Comment by jreback** [comment]
agree this is not pretty. @gerrymanoim can you create an issue to remove this code once `ibis-bigquery` drops support for ibis < 2.0 (which we should release soon anyhow).

**Review by tswast — APPROVED** [review]

**Comment by gerrymanoim** [comment]
https://github.com/ibis-project/ibis-bigquery/issues/62

**Pull request merged**